decidim-ai/lib/decidim/ai/spam_detection/resource/base.rb

```diff
@@ -40,7 +40,11 @@ def error_message(klass, method_name)
           end
 
           def resource_hidden?(resource)
-            resource.class.included_modules.include?(Decidim::Reportable) && resource.hidden?
+            resource.class.included_modules.include?(Decidim::Reportable) && resource.hidden? && spam_reasons.exclude?(resource.reports&.last&.reason)
```

Comment:
Upon checking again the moderation engine, there is no way for an admin link the decision of hiding a resource on a specific report. The engine works more like "I am hiding this resource because of all THOSE reports" ... not a "I am hiding this resource because of THIS report". 
@andra-panaite You can skip this comment.